test/tests/performance/test_dependency_perf.py

```diff
@@ -0,0 +1,83 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2020 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.performance import *
+
+
+class TestDependencyPerformance(TestPerformance):
+    """
+    Test the performance of job dependency feature
+    """
+    def check_depend_delete_msg(self, pjid, cjid):
+        """
+        helper function to check ia message that the dependent job (cjid)
+        is deleted because of the parent job (pjid)
+        """
+        msg = cjid + ";Job deleted as result of dependency on job " + pjid
+        self.server.log_match(msg)
+
+    @timeout(1800)
+    def test_delete_long_dependency_chains(self):
+        """
+        Submit a very long chain of dependent jobs and then measure the time
+        PBS takes to get rid of all dependent jobs.
+        """
+
+        a = {'job_history_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        job = Job()
+        job.set_sleep_time(3600)
+        jid = self.server.submit(job)
+        j_arr = [jid]
+        for each in range(5000):
+            a = {ATTR_depend: 'afternotok:' + jid}
+            jid = self.server.submit(Job(attrs=a))
+            j_arr.append(jid)
+
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=j_arr[0])
+        self.server.expect(JOB, {ATTR_state: 'H'}, id=j_arr[5000])
```

Comment:
Use expexct(JOB, {ATTR_state + '=H': 5000})

Reply:
Isn't it essentially the same thing?

Reply:
If all goes well, yes.  If a bug sneaks in and only some of them are held, then this won't be the same.